docs/extensions/tutorials/_posts/1970-01-05-Installing3rdPartyPackages.md

```diff
@@ -0,0 +1,323 @@
+title: Installing 3rd Party Packages
+section: Tutorials
+---
+
+# Installing 3rd Party Packages
+<hr />
+
+This tutorial will show you how to install a 3rd party package into an extension and through that into the entire app. For an example we're going to use [react-native-swiper](https://github.com/leecade/react-native-swiper) as a non-native package and [react-native-camera](https://github.com/lwansbrough/react-native-camera) as a native package.
+
+## 1) Installing a non-native Package
+
+### Making an Extension
+
+To begin, we'll need an extension, so let's just make one. If this is something you don't know how to do yet, you should really go through [My First Extension]({{ site.url }}/docs/extensions/my-first-extension/introduction) to make sure you understand the fundamentals.
+
+We begin by initializing an extension and writing in the basic information about it.
+
+```ShellSession
+$ shoutem init swiper-extension
+Enter information about your extension. Press `return` to accept (default) values.
+Title: Swiper Extension
+Version: 0.0.1
+Description: Extension that uses react-native-swiper.
+
+Initializing extension:
+...
+
+Extension initialized!
+```
+
+Let's switch over to the extension directory and add a screen with a shortcut that will show the Swiper.
+
+```ShellSession
+$ cd swiper-extension
+$ shoutem screen add SwiperScreen --shortcut Swiper
+Enter shortcut information:
+Title: Swiper
+
+Screen `SwiperScreen` is created in file `app/screens/SwiperScreen.js`!
+Shortcut `Swiper` is created.
+Shortcut `Swiper` opens `SwiperScreen` screen.
+File `app/extension.js` was modified.
+File `extension.json` was modified.
+```
+
+### Installing the Package
+
+There's two different ways you can do this. One is to manually install the package into the extension using `npm install react-native-swiper --save` in the `app` directory:
+
+```ShellSession
+$ cd app
+$ npm install react-native-swiper --save
+{{ site.example.devName }}.swiper-extension@0.0.1 /absolute/path/swiper-extension/app
+└── react-native-swiper@1.5.4
+```
+
+> #### Note
+> The reason we have to install it into the `app` directory is because the the `app` directory is bundled into the full app along with all the other extensions the app uses.
+
+Doing this will automatically add `react-native-swiper` as a dependency in our `app/package.json` file:
+
+```json{5-7}
+#file: app/package.json
+{
+  "name": "{{ site.example.devName }}.not-swiper",
+  "version": "0.0.1",
+  "description": "Extension that uses react-native-swiper.",
+  "dependencies": {
+    "react-native-swiper": "^1.5.4"
+  }
+}
+```
+
+The other, more preferred way of doing this is to simply add that dependency we see in the `app/package.json` file, which results with `react-native-swiper` being installed when the app is bundled. The reason this is the preferred way of doing it is because the `node_modules` folder generated with our NPM installation is ignored since we utilize the `$ npm pack` command.
```

Comment:
Few things here:
- It's not a preferred way to add dependency directly in `app/package.json` because it's usually cumbersome to do so. It's easier to add it with `npm install` with `--save` option.
- i would delete that there are 2 ways and only describe the way with `npm install`.
- when app is bundled -> when extensions are bundled into the app
- remove that "it's preferred way of doing it...", but describe that `node_modules` are ignored by default as extensions is packed with `npm pack` (which ignores it) before installing into the app.